tests/integration_test/transport_security_test/transport_security_test.go

```diff
@@ -149,3 +150,86 @@ func TestServiceControlWithTLS(t *testing.T) {
 		}()
 	}
 }
+
+func TestHttpsClients(t *testing.T) {
+	args := []string{
+		"--service_config_id=test-config-id",
+		"--backend_protocol=http",
+		"--rollout_strategy=fixed",
+		"--server_ssl_path=../../env/testdata",
+	}
+
+	s := env.NewTestEnv(comp.TestHttpsClients, platform.EchoSidecar)
+	defer s.TearDown()
+	s.AppendHttpRules([]*annotationspb.HttpRule{
+		{
+			Selector: "1.echo_api_endpoints_cloudesf_testing_cloud_goog.Simpleget",
+			Pattern: &annotationspb.HttpRule_Get{
+				Get: "/simpleget",
+			},
+		},
+	})
+
+	if err := s.Setup(args); err != nil {
+		t.Fatalf("fail to setup test env, %v", err)
+	}
+
+	testData := []struct {
+		desc         string
+		httpsVersion int
+		certPath     string
+		keyPath      string
```

Comment:
No need for this if key path is not used by client side.

Reply:
Deleted.